diagnostics/app/controllers/DiagnosticsController.scala

```diff
@@ -42,9 +42,12 @@ object DiagnosticsController extends Controller with Logging {
   }
 
   def csp = Action(jsonParser) { implicit request =>
-    if (conf.switches.Switches.CspReporting.isSwitchedOn) {
+    val sampleRate = 0.01
+
+    if (conf.switches.Switches.CspReporting.isSwitchedOn && random.nextDouble <= sampleRate) {
```

Comment:
probably no need to use doubles here, I'd consinder use integers.  might be worth just using the time in milliseconds as it's quick to execute and doing `if (time % 1000 == 1)`